### DISCUSSION TITLE
Consider adding an extra with just httpx

### DISCUSSION THREAD

**Discussion by jonathanberthias** [encode/starlette#2161] [Ideas] [answered]
Hello,

As a user of Starlette, I have had to add `httpx` to my (test) requirements to be able to run tests with the `TestClient`, despite never directly importing anything from `httpx`.

There is already an extra called `full` which would install `httpx`, but that would also install `itsdangerous`, `jinja2`, `python-multipart` and `pyyaml`, none of which I need.

Would you consider adding another extra that would provide only `httpx` with the correct bounds? Perhaps it could be called `testclient` or simply `httpx`.

Thanks!

**Comment by Kludex** [comment, marked as answer]
I don't think it's necessary. Maybe someone created a PR with it in the past, may be worth checking what was the decision back then.
